proposals/PROPOSALS.md

```diff
@@ -0,0 +1,142 @@
+
+The aim of the Haskell Improvement Process is to apply the openness and collaboration that have shaped Haskell's documentation and implementation to a process of evolving the HF and the broader Haskell ecosystem. This document captures our guidelines, commitments and expectations regarding this process.
+
+## Why Write a Proposal?
+
+HIPs are key to making the HF and its initiatives better for the good of everyone. If you decide to invest the time and effort of putting a proposal forward and see it through, your efforts and time will shape and improve the Haskell ecosystem, which means that your proposal may impact the life of a myriad of developers all over the world, including those on your own team. For many, this aspect alone can be quite worthwhile.
+
+However, it's important to note that seeing a proposal through to its conclusion is an involved task. On the one hand, it takes time to convince people that your suggestions are a worthwhile change for hundreds of thousands of developers to accept. Particularly given the sheer volume of developers that could be affected by a proposal, its acceptance is conservative and carefully thought through. Typically, this includes many rounds of discussion with HF leaders, Haskell library maintainers, and the broader community, several iterations on the design of the proposal, and some effort at prototyping the proposed change. Often, it takes weeks to months of discussion, re-design, and prototyping for a proposal to be accepted. It is therefore important to note that seeing a proposal through to its conclusion can be time-consuming and not all proposals may end up accepted, although they may teach us all something!
+
+If you’re motivated enough to go through this involved but rewarding process, go on with writing and keep on reading.
+
+## What's the process for submitting a HIP?
+
+There are four major steps in the HIP process:
+
+1. Initial informal discussion (2 weeks)
+2. Submission
+3. Formal presentation (up to 1 month)
+4. Formal evaluation (up to 2 months)
+
+### Initial informal discussion (2 weeks)
+
+Before submitting a HIP, it is required that you perform necessary preparations:
+
+- Discuss your idea on the [Haskell.org Discourse](https://discourse.haskell.org/). Currently, we suggest cross-posting on the Haskell Foundation Slack for higher volume commentary. Create a Discourse topic under the category "Haskell Foundation", that starts with "Pre-HIP” and briefly describe what you would like to change and why you think it’s a good idea.
+
+- Proposing your ideas on the Discourse is not an optional step. For every change to the ecosystem, it is important to engage in Due Diligence with the community and relevant stakeholders. Use this step to promote your idea and gather early feedback on your Pre-HIP proposal. It may happen that experts and community members may have tried something similar in the past and may offer valuable advice.
```

Comment:
The optional nature of this point is questionable. Perhaps this should be an *internal* HF policy to make sure we don't tread on anyone else's work. In general I don't see a problem with people going straight to submission, though, it may cause more collisions with other people's work than we'd want.

Reply:
Firstly, is there any reason why "Due Diligence" is capitalized here?

As far as the non-optionality of this, if the goal is 'eyeballs' and 'not stepping on other toes', you could mandate _either_ Discourse _or_ Slack? I think that gives some more flexibility, especially for people who prefer a more synchronous discussion medium.

Reply:
> Firstly, is there any reason why "Due Diligence" is capitalized here?

DD is a formal process where I'm from. Must've been a slip. 

> As far as the non-optionality of this, if the goal is 'eyeballs' and 'not stepping on other toes', you could mandate either Discourse or Slack?

If given the choice, I'd choose Discourse, since that's our stated "place to formally interface with the community". The request that the HF also do this on slack could be taken to the extreme: we could do it on Reddit, Twitter, and the mailing lists too. The point is to get the word out that we're interested by casting as wide a net as possible so as to make sure we don't overlap with any existing projects like we did with #7

Reply:
The main reason I felt multiple options would be good is because not everyone is on every platform. To use myself as an example, I'm present on Slack and Twitter and pretty much nowhere else; I'm certain that there are folks who are present on one, and only one, of the options you listed there.

There is some tension here between reach, centralization and convenience; the exact point we want to hit is a little hard to determine.

Reply:
For the most part, I think submission is a fine point to sort out DD. Having a public proposals process before work starts is a good opportunity to figure out what related work exists and how it interacts. In some of the cases where there was overlap, I think having had a better proposals process in place with more public discussion would have helped sort that out earlier and better.

Having pre-discussion on the discourse or whatever is however a good idea as well, since that means that there are proposals that aren't solely one person off on their own, but at least have had some initial scrutiny and feedback before they hit the requirements of a Process.

Reply:
There is a real tension here: while I respect that everyone has different communication preferences, it is also important for there to be one place with the official record of what is happening. Then, an interested observer can monitor the one place. Having additional places is great -- but not at the expense of the one place. Right now, the one place the HF has designated is Discourse.

Another way to think about this is: Who should be inconvenienced more? The proposal authors or (potential) proposal readers? I think the author should bear the brunt of inconvenience: they are proposing a change that will affect potentially the entire Haskell community. This is not to be taken lightly!

As for @gbaz's point: I agree that a proposal can be a fine place to sort out DD, but I think it's not best for HF-sourced proposals, as those can appear to be in advanced stages by the time they hit this repo. See my proposed dichotomy of Proposals vs RFCs below (https://github.com/haskellfoundation/tech-proposals/pull/10#issuecomment-901184121).

Reply:
@goldfirere You make an excellent point - from the point of view of inconveniencing the fewest, having a single, designated location is definitely preferable. I guess the main advantage of Discourse is 'maintained searchable history'?

Reply:
While maintained searchable history is indeed fantastic, my real desire is more about an assumption of asynchronous communication. Discourse's threading model allows someone to come along several days after an announcement and see what they missed. Slack's is potentially capable of this, too, but it's not used in practice: reactions to a comment are often interleaved with new topics. Twitter similarly doesn't offer the organizational tools to be able to easily catch up if one misses a post. Discourse also has an advantage in that it's an open-source tool that we can control directly, in contrast to the other services.

Reply:
:100:% agree with @goldfirere here.